### ISSUE TITLE
Tests are failing?

### ISSUE THREAD

**Issue by andrewculver** [bullet-train-co/bullet_train-super_scaffolding#33]
Was going to cut a release, but something's broken since merging the last couple PRs.

**Comment by gazayas** [comment]
@andrewculver We have this line in [app/views/themes/base/attributes/_date_and_time.html.erb](https://github.com/bullet-train-co/bullet_train-themes/blob/c03bd4c1d609c45bb2b7a8aa87d2de8296840a31/app/views/themes/base/attributes/_date_and_time.html.erb#L7) in `bullet_train-themes`:
```erb
<%= display_date_and_time(object.send(attribute), local_assigns[:date_format], local_assigns[:time_format]) %>
```

This was merged in [bullet_train-themes #6](https://github.com/bullet-train-co/bullet_train-themes/pull/6), but since there were a couple of Joint PRs that went along with this one, we're getting an argument error since the updated `display_date` and `display_date_and_time` methods weren't pulled in at [bullet_train_base #44](https://github.com/bullet-train-co/bullet_train-base/pull/44).

Joint PR → [Tests](https://github.com/bullet-train-co/bullet_train/pull/152)
Joint PR → [Updated methods](https://github.com/bullet-train-co/bullet_train-base/pull/44)

In the meantime, I'll try to address anything else we'll need (i.e. - Deleting the `sed` commands in the Super Scaffolding setup script since we merged #31).

**Comment by gazayas** [comment]
I see, here's the error `main` I'm getting in CircleCI:
```
SuperScaffoldingSystemTest#test_developers_can_generate_a_TestSite_and_a_nested_TestPage_model:
RuntimeError: Wrapped undumpable exception for: ActiveRecord::NotNullViolation: PG::NotNullViolation: ERROR:  null value in column "membership_id" of relation "test_sites" violates not-null constraint
DETAIL:  Failing row contains (1, 1, Some New Example Site, Some Other Value, http://example.org/test, 0, 2022-06-13 10:33:01.462103, 2022-06-13 10:33:01.462103, null).

    app/controllers/account/test_sites_controller.rb:29:in `block in create'
    app/controllers/account/test_sites_controller.rb:28:in `create'


rails test test/system/super_scaffolding_test.rb:47
```

Running this portion of the Super Scaffolding system test works fine in the main application though, so I'll keep looking into why it's failing here.

**Comment by gazayas** [comment]
Still investigating, but here's a short report.

## TL;DR
The `TestSite` test works fine locally in the main app, but the membership select tag doesn't show up at all in CircleCI. 

## In CircleCI
1. Both `@jane` and `@jane.memberships` yield the expected results (In other words, they're not `nil`).
2. Even though `@jane.memberships` is present, the `membership` select tag isn't showing up when creating a `TestSite`. To prove this, when trying to select the membership explicitly, the `select2` tag isn't found:
```
SuperScaffoldingSystemTest#test_developers_can_generate_a_TestSite_and_a_nested_TestPage_model:
Capybara::ElementNotFound: Unable to find css "#select2-test_site_membership_id-container"
    test/system/super_scaffolding_test.rb:60:in `block in <class:SuperScaffoldingSystemTest>'
```
3. Changing the setup for browser tools in CircleCI's `config.yml` didn't have an affect:
```yml
steps:
  - browser-tools/install-chrome
  - browser-tools/install-chromedriver
```

# Locally
1. This test passes even with `CI=true bundle exec rails test test/system/super_scaffolding_system_test.rb`

## Eager loading
Also, the way we're eager loading in the main app doesn't give us any issues there, but it does here, so it turns out we need BOTH methods, not just one or the other, to make things work in this repo as well:
```ruby
config.eager_load = ENV["CI"].present?
Rails.application.eager_load! if ENV["CI"].present?
```
